docs/releases.md

```diff
@@ -1,39 +1,157 @@
+
+- [ ] Final preparation
+  - [ ] Verify that version string in `repo/version.go` has been updated
+  - [ ] tag commit with vX.Y.Z
+  - [ ] update release branch to point to release commit (`git merge vX.Y.Z`).
+  - [ ] publish dist.ipfs.io
+  - [ ] publish next version to https://github.com/ipfs/npm-go-ipfs
+- [ ] Publish a Release Blog post (at minimum, a c&p of this release issue with all the highlights, API changes, link to changelog and thank yous)
+- [ ] Broadcasting (link to blog post)
+  - [ ] Twitter
+  - [ ] IRC
+  - [ ] Reddit
+  - [ ] [discuss.ipfs.io](https://discuss.ipfs.io/c/announcements)
+  - [ ] Announce it on the [IPFS Users mlist](https://groups.google.com/forum/#!forum/ipfs-users)
+
+# ❤️ Huge thank you to everyone that made this release possible
+
+In alphabetical order, here are all the humans that contributed to the release:
+
+- <use script -- listed in Release Stage 4 -- to generate a list of everyone that contributed for this release>
+
+# 🙌🏽 Want to contribute?
+
+Would you like to contribute to the IPFS project and don't know how? Well, there are a few places you can get started:
+
+- Check the issues with the `help wanted` label in the [go-ipfs repo](https://github.com/ipfs/go-ipfs/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)
+- Join an IPFS All Hands, introduce yourself and let us know where you would like to contribute - https://github.com/ipfs/team-mgmt/#weekly-ipfs-all-hands
+- Hack with IPFS and show us what you made! The All Hands call is also the perfect venue for demos, join in and show us what you built
+- Join the discussion at http://discuss.ipfs.io/ and help users finding their answers.
+- Join the [Go Core Dev Team Weekly Sync](https://github.com/ipfs/team-mgmt/issues/650) and be part of the Sprint action!
+
+# ⁉️ Do you have questions?
+
+The best place to ask your questions about IPFS, how it works and what you can do with it is at [discuss.ipfs.io](http://discuss.ipfs.io). We are also available at the `#ipfs` channel on Freenode.
+```
+
+## Release Version Numbers (aka semver)
+
+Until `go-ipfs` 0.4.X, `go-ipfs` was not using semver to communicate the type of release
+
+Post `go-ipfs` 0.5.X, `go-ipfs` will use semver. This means that patch releases will not contain any breaking changes nor new features. Minor releases might contain breaking changes and always contain some new feature
+
+Post `go-ipfs` 1.X.X (future), `go-ipfs` will use semver. This means that only major releases will contain breaking changes, minors will be reserved for new features and patches for bug fixes.
+
+We do not yet retroactively apply fixes to older releases (no Long Term Support releases for now), which means that we always recommend users to update to the latest, whenever possible.
```

Comment:
Ho, bummer...

Please consider this in the future for disruptive problems. At Infura we have been stuck with problematic release for months without solution (other than forking): 0.4.19 and 0.4.20 had memory usage problem but we couldn't revert to 0.4.18 due to security reason. This will only become more critical in the future.

Reply:
In that case, we would have applied the CPU fixes in 0.4.19 to 0.4.18 as a patch. This is just saying that we're not going to release a 0.4.20 and then go back and patch a 0.4.18 release.